core/toolbox/toolbox.js

```diff
@@ -0,0 +1,463 @@
+
+  /**
+   * Array holding info needed to unbind events.
+   * Used for disposing.
+   * Ex: [[node, name, func], [node, name, func]].
+   * @type {!Array<!Array<?>>}
+   * @protected
+   */
+  this.boundEvents_ = [];
+};
+
+/**
+ * Initializes the toolbox
+ * @public
+ */
+Blockly.Toolbox.prototype.init = function() {
+  var workspace = this.workspace_;
+  var svg = workspace.getParentSvg();
+
+  this.flyout_ = this.createFlyout_();
+
+  this.HtmlDiv = this.createDom_(this.workspace_);
+  this.render(this.toolboxDef_);
+  var themeManager = workspace.getThemeManager();
+  themeManager.subscribe(this.HtmlDiv, 'toolboxBackgroundColour',
+      'background-color');
+  themeManager.subscribe(this.HtmlDiv, 'toolboxForegroundColour', 'color');
+
+  // Insert the flyout after the workspace.
+  Blockly.utils.dom.insertAfter(this.flyout_.createDom('svg'), svg);
+  this.flyout_.init(workspace);
+};
+
+/**
+ * Creates the dom for the toolbox.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace this toolbox is on.
+ * @return {!Element} The html container for the toolbox.
+ * @protected
+ */
+Blockly.Toolbox.prototype.createDom_ = function(workspace) {
+  var svg = workspace.getParentSvg();
+
+  var container = this.createContainer_();
+
+  this.contentsDiv_ = this.createContentsContainer_();
+  this.contentsDiv_.tabIndex = 0;
+  Blockly.utils.aria.setRole(this.contentsDiv_, Blockly.utils.aria.Role.TREE);
+  container.appendChild(this.contentsDiv_);
+
+  svg.parentNode.insertBefore(container, svg);
+
+  this.addToolboxListeners_(container, this.contentsDiv_);
+  return container;
+};
+
+/**
+ * Creates the container div for the toolbox.
+ * @return {!Element} The html container for the toolbox.
+ * @protected
+ */
+Blockly.Toolbox.prototype.createContainer_ = function() {
+  var toolboxContainer = document.createElement('div');
+  toolboxContainer.setAttribute('layout', this.isHorizontal() ? 'h' : 'v');
+  Blockly.utils.dom.addClass(toolboxContainer, 'blocklyToolboxDiv');
+  Blockly.utils.dom.addClass(toolboxContainer, 'blocklyNonSelectable');
+  toolboxContainer.setAttribute('dir', this.RTL ? 'RTL' : 'LTR');
+  return toolboxContainer;
+};
+
+/**
+ * Creates the container for all the contents in the toolbox.
+ * @return {!Element} The html container for the toolbox contents.
+ * @protected
+ */
+Blockly.Toolbox.prototype.createContentsContainer_ = function() {
+  var contentsContainer = document.createElement('div');
+  Blockly.utils.dom.addClass(contentsContainer, 'blocklyToolboxContents');
+  if (this.isHorizontal()) {
+    contentsContainer.style.flexDirection = 'row';
+  }
+  return contentsContainer;
+};
+
+/**
+ * Adds event listeners to the toolbox container div.
+ * @param {!Element} container The html container for the toolbox.
+ * @param {!Element} contentsContainer The html container for the contents
+ *     of the toolbox.
+ * @protected
+ */
+Blockly.Toolbox.prototype.addToolboxListeners_ = function(container,
```

Comment:
How about `attachEvents_` instead? It'd be great if we can align on one method similar to `init` and `render` that applies to all types of widgets.

Reply:
done.